File: 0342-power-of-four/0342-power-of-four.py
class Solution(object):
    def isPowerOfFour(self, n):
        """
        :type n: int
        :rtype: bool
        """
        max_i = 2**31 - 1
        k = int(math.log(max_i, 4))
        largestP = 4 ** k
        return n > 0 and (largestP % n == 0) and math.log(n, 4).is_integer()

File: 0342-power-of-four/0342-power-of-four.java
class Solution {
    public boolean isPowerOfFour(int n) {
        if(n <= 0){
            return false;

        }

        int max_i = (int) Math.pow(2,31) -1;
        int k = (int) (Math.log(max_i)/ Math.log(4));
        long LargestP =(long) Math.pow(4, k);

        return (LargestP % n == 0) && (Math.log(n)/ Math.log(4) % 1 == 0);
        }}